src/main/java/io/opentracing/contrib/concurrent/TracedExecutor.java

```diff
@@ -28,20 +28,21 @@ public TracedExecutor(Executor executor, Tracer tracer, boolean traceWithActiveS
 
   @Override
   public void execute(Runnable runnable) {
-    Scope scope = createScope("execute");
+    Span span = createSpan("execute");
     try {
-      delegate.execute(tracer.activeSpan() == null ? runnable :
-          new TracedRunnable(runnable, tracer));
+      Span toActivate = span != null ? span : tracer.activeSpan();
+      delegate.execute(toActivate == null ? runnable : new TracedRunnable(runnable, tracer, toActivate));
     } finally {
-      if (scope != null) {
-        scope.close();
+      // close the span if created
+      if (span != null) {
+        span.finish();
       }
     }
   }
 
-  Scope createScope(String operationName) {
+  Span createSpan(String operationName) {
```

Comment:
Nit: The method name does not indicate that there may _not_ be a span created (which in cases with active spans would actually be the norm).

The name should make it feel natural that `null` gets returned and in which situations. Something like `createNewSpanWhenNoActiveSpanExists()` may be a bit too verbose, but I hope you get my point.
This rename may be a good time to think of a better name.

Reply:
thanks @sjoerdtalsma I will leave it as it was. The method also takes into account an internal parameter whether to create a span or not.